src/jqLite.js

```diff
@@ -313,17 +313,23 @@ function jqLiteOff(element, type, fn, unsupported) {
       delete events[type];
     }
   } else {
-    forEach(type.split(' '), function(type) {
+
+    var removeHandler = function(type) {
+      var listenerFns = events[type];
       if (isDefined(fn)) {
-        var listenerFns = events[type];
         arrayRemove(listenerFns || [], fn);
-        if (listenerFns && listenerFns.length > 0) {
-          return;
-        }
       }
+      if (!(isDefined(fn) && listenerFns && listenerFns.length > 0)) {
```

Comment:
I think that this construct is correct, since we call `removeEventListenerFn` in the following cases:

* `fn` is not defined - i.e. we want to remove all handlers for this event
* `fn` is defined AND there are no more items in the `listenerFns` array